p2p/src/authenticated/discovery/actors/peer/actor.rs

```diff
@@ -187,10 +184,11 @@ impl<E: Spawner + Clock + ReasonablyRealtime + Rng + CryptoRng + Metrics, C: Pub
                         .map_err(Error::ReceiveFailed)?;
 
                     // Parse the message
-                    let msg = match types::Payload::decode_cfg(msg, &self.codec_config) {
+                    let cfg = (msg.len(), self.peer_gossip_max_count);
```

Comment:
This allows us to use the `msg.len()` to bound what we pre-allocate into the inner vector

Reply:
Need to find some way to comment this...